eodag/plugins/download/http.py

```diff
@@ -908,7 +876,7 @@ def _stream_download(
         auth: Optional[AuthBase] = None,
         progress_callback: Optional[ProgressCallback] = None,
         **kwargs: Unpack[DownloadConf],
-    ) -> Iterator[Any]:
+    ) -> Union[Callable[[], Any], None]:
```

Comment:
was the `iteration_wrapper` needed ? It changes the method output, and is no more correlated to `AwsDownload._stream_download` and https://eodag.readthedocs.io/en/latest/plugins_reference/download.html#download-methods-call-graph